src/main/java/io/cdap/plugin/snowflake/common/SnowflakeErrorDetailsProvider.java

```diff
@@ -0,0 +1,161 @@
+      if (t instanceof URISyntaxException) {
+        return getProgramFailureException((URISyntaxException) t, errorContext);
+      }
+      if (t instanceof SchemaParseException) {
+        return getProgramFailureException((SchemaParseException) t, errorContext);
+      }
+      if (t instanceof UnexpectedFormatException) {
+        return getProgramFailureException((UnexpectedFormatException) t, errorContext);
+      }
+      if (t instanceof ConnectionTimeoutException) {
+        return getProgramFailureException((ConnectionTimeoutException) t, errorContext);
+      }
+    }
+    return null;
+  }
+
+  /**
+   * Get a ProgramFailureException with the given error
+   * information from {@link IllegalArgumentException}.
+   *
+   * @param e The IllegalArgumentException to get the error information from.
+   * @return A ProgramFailureException with the given error information.
+   */
+  private ProgramFailureException getProgramFailureException(IllegalArgumentException e, ErrorContext errorContext) {
+    String errorMessage = e.getMessage();
+    String errorMessageFormat = "Error occurred in the phase: '%s'. Error message: %s";
+
+    return ErrorUtils.getProgramFailureException(new ErrorCategory(ErrorCategory.ErrorCategoryEnum.PLUGIN),
+      errorMessage,
+      String.format(errorMessageFormat, errorContext.getPhase(), errorMessage), ErrorType.USER, false, e);
+  }
+
+  /**
+   * Get a ProgramFailureException with the given error
+   * information from {@link IllegalStateException}.
+   *
+   * @param e The IllegalStateException to get the error information from.
+   * @return A ProgramFailureException with the given error information.
+   */
+  private ProgramFailureException getProgramFailureException(IllegalStateException e, ErrorContext errorContext) {
+    String errorMessage = e.getMessage();
+    String errorMessageFormat = "Error occurred in the phase: '%s'. Error message: %s";
+    return ErrorUtils.getProgramFailureException(new ErrorCategory(ErrorCategory.ErrorCategoryEnum.PLUGIN),
+      errorMessage,
+      String.format(errorMessageFormat, errorContext.getPhase(), errorMessage), ErrorType.SYSTEM, false, e);
+  }
+
+  /**
+   * Get a ProgramFailureException with the given error
+   * information from {@link URISyntaxException}.
+   *
+   * @param e The URISyntaxException to get the error information from.
+   * @return A ProgramFailureException with the given error information.
+   */
+  private ProgramFailureException getProgramFailureException(URISyntaxException e,
+                                                             ErrorContext errorContext) {
+    String errorMessage = e.getMessage();
+    String errorMessageFormat = "Error occurred in the phase: '%s'. Error message: %s";
+    return ErrorUtils.getProgramFailureException(new ErrorCategory(ErrorCategory.ErrorCategoryEnum.PLUGIN),
+      errorMessage,
+      String.format(errorMessageFormat, errorContext.getPhase(), errorMessage), ErrorType.SYSTEM, false, e);
```

Comment:
can you please provide rationale as comment, why all these exceptions are considered as `SYSTEM` error?

Reply:
I did not ask to convert them to `USER`, question was to provide why these are marked as SYSTEM errors?

Reply:
`URISyntaxException` is not being thrown in this plugin, i think we can remove this case.
For classification all error due to user input can be classified as `USER`

Reply:
removed this case from detail provider, as this was only catched, but wrapped under IllegalArgumentException